### PR TITLE
Feature 18 vacation request status object

### PR DESCRIPTION
**Pull request by VeikkoVilpponen**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -491,6 +492,176 @@ paths:
...
             type: string
             format: uuid
+        - name: personId
+          required: true
+          in: path
+          description: person id
+          schema:
+            type: integer
+            format: int32
+
```
Comment (Jdallos) [inline swagger.yaml]:
```suggestion
```

File: swagger.yaml [hunk 2]
```diff
@@ -477,6 +477,7 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/Error"
+  /v1/vacationsRequests/{id}/{personId}:
```
Comment (Jdallos) [inline swagger.yaml]: So previously a vacation request was deleted by the vacation request id, now it also requires the personId. I assume there is a reason for this?

File: swagger.yaml [hunk 3]
```diff
@@ -491,6 +492,176 @@ paths:
...
+      operationId: listVacationRequestStatuses
+      summary: List vacation requests statuses
+      description: List vacation requests statuses
+      tags:
+        - VacationRequestStatus
+      parameters:
+        - name: vacationRequestId
+          required: false
+          in: query
+          description: Filter by persons unique ID
```
Comment (Jdallos) [inline swagger.yaml]:
```suggestion
          description: Filter by vacation requests unique ID
```

File: swagger.yaml (hunk 2 above)
Comment (Jdallos) [inline swagger.yaml]:
```suggestion

  /v1/vacationsRequests/{id}/{personId}:
```

File: swagger.yaml [hunk 4]
```diff
@@ -835,26 +1004,47 @@ components:
...
-          description: ID of person who modified request last
         updatedAt:
           type: string
           format: date-time
           description: Date time of vacation request update
-        lastUpdatedBy:
+
+    VacationRequestStatus:
+      type: object
+      description: singular vacation request status chance
```
Comment (Jdallos) [inline swagger.yaml]: chance?

File: swagger.yaml [hunk 5]
```diff
@@ -835,26 +1004,47 @@ components:
...
+        vacationRequestId:
+          type: string
+          format: uuid
+          description: ID of vacation request in UUID
+          default: null
+        person:
+          type: integer
+          format: int32
+          description: Person ID in Forecast
+        status:
```
Comment (Jdallos) [inline swagger.yaml:1041]: Would this be better as statuses?

**Review by Jdallos — COMMENTED** [review]
Some comments and suggestions, will leave approval to Ville

File: swagger.yaml [hunk 6]
```diff
@@ -815,9 +989,9 @@ components:
           description: ID of vacation request in UUID
           default: null
         person:
```
Comment (villejuutila) [inline swagger.yaml]:
```suggestion
        personId:
```

File: swagger.yaml [hunk 7]
```diff
@@ -815,9 +989,9 @@ components:
...
-          type: integer
-          format: int32
-          description: Person ID in Forecast
+          type: string
+          format: uuid
+          description: Person ID in UUID
```
Comment (villejuutila) [inline swagger.yaml]:
```suggestion
          description: Persons Keycloak ID in UUID
```

**Review by villejuutila — APPROVED** [review]
Approved

File: swagger.yaml (hunk 5 above)
Comment (VeikkoVilpponen) [reply swagger.yaml:1041]: I think it is better as it is sense it is just one status at a time. The enum contains all statuses so I think the name fits it.

**Pull request merged**